### PR TITLE
Add using casing check/fix for initMountPoints

### PR DESCRIPTION
**Pull request by MorrisJobke**
- downstream of https://github.com/owncloud/core/pull/26398 and https://github.com/owncloud/core/pull/26464

@icewind1991 Could you have a look at this, because it involves filesystem stuff

cc @rullzer @nickvergessen

### CODE REVIEW

**Comment by MorrisJobke** [comment]
I tested this code and at least my tests didn't revealed anything bad.

File: lib/private/Share/Share.php [hunk 1]
```diff
@@ -136,6 +136,15 @@ public static function isEnabled() {
 	 *       not '/admin/data/file.txt'
 	 */
 	public static function getUsersSharingFile($path, $ownerUser, $includeOwner = false, $returnUserPaths = false, $recursive = true) {
+		$userManager = \OC::$server->getUserManager();
+		$userObject = $userManager->get($ownerUser);
+
+		if (is_null($ownerUser)) {
+			\OCP\Util::writeLog('files', ' Backends provided no user object for ' . $ownerUser, \OCP\Util::ERROR);
+			throw new \OC\User\NoUserException('Backends provided no user object for ' . $ownerUser);
```
Comment (LukasReschke) [inline lib/private/Share/Share.php]: I'd prefer not adding potential user controlled content to the exception.

Comment (LukasReschke) [inline lib/private/Share/Share.php]: … that probably too … or be below L147. Good catch ❗️

Comment (MorrisJobke) [inline lib/private/Share/Share.php]: > be below L147. Good catch

This makes no sense, because if the userObject is null then the method will crash. 😉

Comment (MorrisJobke) [inline lib/private/Share/Share.php]: yep - that's the reason why @nickvergessen comment is correct ;)

Comment (LukasReschke) [inline lib/private/Share/Share.php]: Ah. Gotcha. I thought you referred to our comments 🙈

**Comment by MorrisJobke** [comment]
@nickvergessen @LukasReschke I fixed the issues. Please review again 🤔

File: lib/private/Share/Share.php [hunk 2]
```diff
@@ -136,6 +136,15 @@ public static function isEnabled() {
 	 *       not '/admin/data/file.txt'
 	 */
 	public static function getUsersSharingFile($path, $ownerUser, $includeOwner = false, $returnUserPaths = false, $recursive = true) {
+		$userManager = \OC::$server->getUserManager();
```
Comment (nickvergessen) [inline lib/private/Share/Share.php]: DI ?

Comment (MorrisJobke) [inline lib/private/Share/Share.php]: Static code 🙈

File: lib/private/Files/Node/Root.php [hunk 3]
```diff
@@ -347,6 +347,15 @@ public function getName() {
 	 * @return \OCP\Files\Folder
 	 */
 	public function getUserFolder($userId) {
+		$userObject = \OC::$server->getUserManager()->get($userId);
```
Comment (LukasReschke) [inline lib/private/Files/Node/Root.php]: DI

File: lib/private/Files/Node/Root.php [hunk 4]
```diff
@@ -347,6 +347,15 @@ public function getName() {
...
+
+		if (is_null($userObject)) {
+			\OCP\Util::writeLog('files', 'Backends provided no user object for ' . $userId, \OCP\Util::ERROR);
```
Comment (LukasReschke) [inline lib/private/Files/Node/Root.php]: DI

File: lib/private/Share/Share.php (hunk 2 above)
Comment (LukasReschke) [inline lib/private/Share/Share.php]: Let me also look into that…

File: lib/private/Share/Share.php [hunk 5]
```diff
@@ -136,6 +136,15 @@ public static function isEnabled() {
...
+		$userObject = $userManager->get($ownerUser);
+
+		if (is_null($userObject)) {
+			\OCP\Util::writeLog('files', ' Backends provided no user object for ' . $ownerUser, \OCP\Util::ERROR);
```
Comment (LukasReschke) [inline lib/private/Share/Share.php]: DI

**Review by LukasReschke — DISMISSED** [review]
Let me try to do some changes here to make the code cleaner.

**Comment by LukasReschke** [comment]
@MorrisJobke @nickvergessen Did some changes so we have proper DI. Best to go through commits. Please review. 

LGTM for me on the original changes.

Also cc @icewind1991 since it's filesystem stuff.

**Comment by MorrisJobke** [comment]
:+1: for @LukasReschke changes

**Comment by blizzz** [comment]
😭

**Pull request merged**